main.tf

```diff
@@ -77,6 +81,14 @@ provider "aws" {
   }
 }
 
+provider "snowflake" {
+  role  = "ACCOUNTADMIN"
+  username = var.snowflake_username
```

Comment:
Are these in Vault? If so, you should read them from vault Directly.

Reply:
Right now I have the snowflake_private_key as a sensitive variable in a terraform cloud variable set, but I could change that and the others to vault.

Reply:
@Eric-Fontana-Indico I don't think we'll be able to pull these from vault, as the the provider needs them to authenticate first, before we can even get them from vault.  Even if we put the provider block inside the snowflake module, we can't use a depends_on to authenticate vault first because " The module at module.snowflake is a legacy module which contains its own
│ local provider configurations, and so calls to it may not use the count,
│ for_each, or depends_on arguments." I don't think having the snowflake_private_key variable in each cluster would break anything, though, as it shouldn't affect any other thing in the cluster, it would just be an unused env var, I would think.